File: java-workshop/part-1-powermock/src/main/java/infrastructure/PaymentGateway.java
package infrastructure;

import domain.PaymentResult;
import domain.PaymentStatus;
import java.util.UUID;

/**
 * Static methods - requires PowerMock to mock.
 */
public final class PaymentGateway {
    private PaymentGateway() {
    }

    public static PaymentResult processPayment(String customerId, double amount) {
        if (amount > 0) {
            return new PaymentResult("", PaymentStatus.SUCCESS, UUID.randomUUID().toString());
        }
        return new PaymentResult("", PaymentStatus.FAILED, null);
    }

    public static boolean refund(String transactionId) {
        return true;
    }
}

File: kotlin-workshop/part-1-powermock/src/main/kotlin/infrastructure/PaymentGateway.kt
package infrastructure

import domain.PaymentResult
import domain.PaymentStatus
import java.util.UUID

/**
 * Static methods - another thing that requires PowerMock-style mocking
 */
object PaymentGateway {
    
    fun processPayment(customerId: String, amount: Double): PaymentResult {
        // In real life, this would call an external payment provider
        return if (amount > 0) {
            PaymentResult(
                orderId = "",
                status = PaymentStatus.SUCCESS,
                transactionId = UUID.randomUUID().toString()
            )
        } else {
            PaymentResult(
                orderId = "",
                status = PaymentStatus.FAILED
            )
        }
    }
    
    fun refund(transactionId: String): Boolean {
        // Simulate refund
        return true
    }
}

